stingray/io.py

```diff
@@ -944,31 +948,37 @@ def _initialize_header_events(self, fname, force_hduname=None):
             hduname = 1
         self._add_meta_attr("hduname", hduname)
 
-        self._add_meta_attr("header", dict(hdulist[self.hduname].header))
-        self._add_meta_attr("nphot", self.header["NAXIS2"])
+        header = hdulist[hduname].header
+
+        # self.header has to be a string, for backwards compatibility and... for convenience!
```

Comment:
I think these lines are exhaustive and describe the changes needed. I do not have anything to report.